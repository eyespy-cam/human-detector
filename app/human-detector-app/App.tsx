import * as React from 'react';
import { NavigationContainer } from '@react-navigation/native';
import { createNativeStackNavigator } from '@react-navigation/native-stack';
import { Device } from 'react-native-ble-plx';
import CameraScreen from './screens/CameraScreen';
import GroupScreen from './screens/GroupScreen';
import LoginScreen from './screens/LoginScreen';
import BluetoothScreen from './screens/CameraRegistration/BluetoothScreen';
import LoadingScreen from './screens/CameraRegistration/LoadingScreen';
import EnterCameraRegInfoScreen from './screens/CameraRegistration/EnterCameraRegInfoScreen';
import fetchPushToken from './src/notifications/fetchPushToken';
import BackendService from './services/backendService';
import { BackendContext } from './contexts/backendContext';

const Stack = createNativeStackNavigator();

export default function App(): React.ReactElement {
  const [backendService, setBackendService] = React.useState<BackendService | null>(null);
  const [currentDevice, setConnectedDevice] = React.useState<Device | null>(null);

  const bleContext = {
    device: currentDevice,
    setDevice: setConnectedDevice,
    connect: async (device: Device) => {
     try {
       const deviceConnection = await device.connect();
       setConnectedDevice(deviceConnection);
     } catch (error) {
       console.log('ERROR WHEN CONNECTING', error);
     }
   }
  }

  // If the user isn't logged in
  if (backendService === null) {
    // Open the login screen
    return (
      <NavigationContainer>
        <Stack.Navigator>
          <Stack.Screen
            name="Login"
            options={{
              title: 'EyeSpy',
              headerStyle: {
                backgroundColor: '#1E90FF',
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold',
              },
            }}
          >
            {() => (
              <LoginScreen
                onSuccessfulLogin={(tokenManager) => {
                  const backend = new BackendService(tokenManager);
                  setBackendService(backend);
                  fetchPushToken()
                    .then((token) => backend.sendNotifyTokenAPI(token))
                    .catch((error) =>
                      console.error('Error fetching or updating push token', error)
                    );
                }}
              />
            )}
          </Stack.Screen>
        </Stack.Navigator>
      </NavigationContainer>
    );
  }

  // If the user is logged in return the stack with all the information
  return (
    <NavigationContainer>
      <BackendContext.Provider value={backendService}>
        <Stack.Navigator>
          <Stack.Screen
            name="Groups"
            component={GroupScreen}
            options={{
              title: 'EyeSpy',
              headerStyle: {
                backgroundColor: '#1E90FF',
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold',
              },
            }}
          />
          <Stack.Screen
            name="Cameras"
            component={CameraScreen}
            options={{
              title: 'EyeSpy',
              headerStyle: {
                backgroundColor: '#1E90FF',
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold',
              },
              headerBackTitleVisible: false,
              headerBackVisible: true,
            }}
          />
          <Stack.Screen
            name="Bluetooth"
            component={BluetoothScreen}
            options={{
              title: 'Connect Camera',
              headerStyle: {
                backgroundColor: '#1E90FF',
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold',
              },
            }}
          />
          <Stack.Screen
            name="CameraRegistrationInfo"
            component={EnterCameraRegInfoScreen}
            options={{
              title: 'Connect Camera',
              headerStyle: {
                backgroundColor: '#1E90FF',
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold',
              },
              headerBackTitleVisible: true,
              headerBackVisible: true,
            }}
          />
          <Stack.Screen
            name="Loading"
            component={LoadingScreen}
            options={{
              title: 'Connect Camera',
              headerStyle: {
                backgroundColor: '#1E90FF',
              },
              headerTintColor: '#fff',
              headerTitleStyle: {
                fontWeight: 'bold',
              },
              headerBackTitleVisible: true,
              headerBackVisible: true,
            }}
          />
        </Stack.Navigator>
      </BackendContext.Provider>
    </NavigationContainer>
  );
}
