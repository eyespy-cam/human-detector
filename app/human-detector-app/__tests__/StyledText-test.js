import * as React from 'react';
import renderer from 'react-test-renderer';
import { StyleSheet, Text, View, Button, Alert } from 'react-native';
import App from '../App';

//Group Button
it(`renders button correctly`, () => {
  let test = 3
  expect(test == 4);
});

it(`group button clicks correctly`, () => {
  expect(14 == 3);
});

it(`test button`, () => {
  const component = renderer.create(<DisplayCameraButton></DisplayCameraButton>)
  expect(component.toJSON()).toMatchSnapshot();
});
