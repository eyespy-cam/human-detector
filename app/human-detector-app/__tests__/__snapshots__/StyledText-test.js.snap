// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button correctly 1`] = `
<View
  accessibilityRole="button"
  accessible={true}
  collapsable={false}
  focusable={true}
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Array [
        Object {},
      ]
    }
  >
    <Text
      style={
        Array [
          Object {
            "color": "#007AFF",
            "fontSize": 18,
            "margin": 8,
            "textAlign": "center",
          },
        ]
      }
    >
      Test test
    </Text>
  </View>
</View>
`;
